llvm/lib/Target/AMDGPU/AsmParser/AMDGPUAsmParser.cpp

```diff
@@ -4006,6 +4007,29 @@ bool AMDGPUAsmParser::validateMIMGGatherDMask(const MCInst &Inst) {
   return DMask == 0x1 || DMask == 0x2 || DMask == 0x4 || DMask == 0x8;
 }
 
+bool AMDGPUAsmParser::validateMIMGDim(const MCInst &Inst,
+                                      const OperandVector &Operands) {
+  if (!isGFX10Plus())
```

Comment:
can this be a named feature instead of a generation check

Reply:
I was not able to find a named feature for "dim". Note that `isGFX10Plus()` etc are defined as member functions of the class `AMDGPUAsmParser` and are used throughout the file. For example, the first few lines of `parseDim()` include a call of `isGFX10Plus()`:
```c++
ParseStatus AMDGPUAsmParser::parseDim(OperandVector &Operands) {
  if (!isGFX10Plus())
    return ParseStatus::NoMatch;
...
```